larq_compute_engine/mlir/transforms/optimize_patterns_target_arm.td

```diff
@@ -0,0 +1,20 @@
+include "larq_compute_engine/mlir/transforms/optimize_patterns_common.td"
+
+// Insert a binary maxpool if a maxpool is followed by a sign op.
+def : Pat<(LQ_QuantizeOp
+              (TFL_MaxPool2DOp:$pool_output
+                  $input,
+                  $padding,
+                  $stride_w,
+                  $stride_h,
+                  $filter_width,
+                  $filter_height,
+                  $fused_activation_function)),
+          (LQ_BMaxPool2dOp
+              (LQ_QuantizeOp $input),
+              $padding,
+              $stride_w,
+              $stride_h,
+              $filter_width,
+              $filter_height),
+          [(HasOneUse $pool_output)]>;
```

Comment:
The format here is that there is one `optimize_patterns_common.td` file that defines the common re-write patterns, and is imported and extended as needed by target-specific table-gen files, as is being done here.